stdlib/src/collections/list.mojo

```diff
@@ -497,12 +497,72 @@ struct List[T: CollectionElement, hint_trivial_type: Bool = False](
 
         Args:
             value: The value to append.
+
+        Notes:
+            If there is no capacity left, resizes to twice the current capacity.
+            Except for 0 capacity where it sets 1.
         """
         if self.size >= self.capacity:
-            self._realloc(max(1, self.capacity * 2))
+            self._realloc(self.capacity * 2 + int(self.capacity == 0))
         (self.data + self.size).init_pointee_move(value^)
         self.size += 1
 
+    fn append[
+        D: DType, //
+    ](mut self: List[Scalar[D], *_, **_], value: SIMD[D, _]):
+        """Appends a vector to this list.
+
+        Parameters:
+            D: The DType.
+
+        Args:
+            value: The value to append.
+
+        Notes:
+            If there is no capacity left, resizes to `len(self) + value.size`.
+        """
+        self.reserve(self.size + value.size)
+        (self.data + self.size).store(value)
+        self.size += value.size
+
+    fn append[
+        D: DType, //
+    ](mut self: List[Scalar[D], *_, **_], value: SIMD[D, _], count: Int):
+        """Appends a vector to this list.
+
+        Parameters:
+            D: The DType.
+
+        Args:
+            value: The value to append.
+            count: The ammount of items to append.
+
+        Notes:
+            If there is no capacity left, resizes to `len(self) + count`.
+        """
+        self.reserve(self.size + count)
+        var v_ptr = UnsafePointer.address_of(value).bitcast[Scalar[D]]()
+        memcpy(self.data + self.size, v_ptr, count)
```

Comment:
**Suggestion** For this method to be safe, I think we need a check that `count` is not larger than the SIMD width of `value`. Otherwise, someone might be able to read more bytes from `value` than it actually contains, causing UB.

```suggestion
        memcpy(self.data + self.size, v_ptr, count)
        debug_assert(count <= value.size, "cannot append more elements than appear in SIMD vector")
```

Reply:
Great catch